presto-main/src/main/java/com/facebook/presto/server/remotetask/TaskInfoFetcher.java

```diff
@@ -209,9 +215,14 @@ private synchronized void sendNextRequest()
 
         HttpUriBuilder httpUriBuilder = uriBuilderFrom(taskStatus.getSelf());
         URI uri = summarizeTaskInfo ? httpUriBuilder.addParameter("summarize").build() : httpUriBuilder.build();
-        Request request = setContentTypeHeaders(isBinaryTransportEnabled, prepareGet())
-                .setUri(uri)
-                .build();
+        Request.Builder uriBuilder = setContentTypeHeaders(isBinaryTransportEnabled, prepareGet());
+
+        if (taskInfoRefreshMaxWait.toMillis() > 0) {
+            uriBuilder.setHeader(PRESTO_CURRENT_STATE, taskStatus.getState().toString())
+                    .setHeader(PRESTO_MAX_WAIT, taskInfoRefreshMaxWait.toString());
```

Comment:
Currently the way TaskResource is implemented - as long as the header is set (even to 0s) - it creates features, and does more things than if the header is not set.

I'm not trying to imply that this is a problem, but it feels like it makes some sense to make sure that by default it has no additional implications. I would check for 0 here, and if the delay is 0 - don't set the header.